### DISCUSSION TITLE
Is it considered acceptable to directly access cached data using queryClient.getQueryData within a React component?

### DISCUSSION THREAD

**Discussion by shehan-sandaruwan** [TanStack/query#6239] [Q&A]
I'm actively involved in a React project that utilizes react-query, and I've encountered a situation where I must update real-time data obtained via a WebSocket connection. Since we've opted not to use a state management library like Redux, we're leveraging the capabilities provided by react-query.

My approach involves directly updating the query cache when WebSocket data arrives. My primary concern is whether it's considered acceptable to access query cache data directly within a React component. This allows me to trigger component re-renders and present the user with the most up-to-date information.

**Comment by kalijonn** [comment]
Would this help?

https://tkdodo.eu/blog/using-web-sockets-with-react-query

**Comment by TkDodo** [comment]
> My primary concern is whether it's considered acceptable to access query cache data directly within a React component.

you always want `useQuery` for that, but you can have a read-only observer by setting `enabled:false`
